tests/main/test_agent.py

```diff
@@ -1,72 +1,70 @@
 import asyncio
+from typing import Optional
 
 import pytest
 
-from langroid.agent.base import Agent, AgentConfig
-from langroid.cachedb.redis_cachedb import RedisCacheConfig
-from langroid.language_models.base import StreamingIfAllowed
-from langroid.language_models.openai_gpt import OpenAIChatModel, OpenAIGPTConfig
-from langroid.parsing.parser import ParsingConfig
-from langroid.prompts.prompts_config import PromptsConfig
-from langroid.utils.configuration import Settings, set_global
-from langroid.vector_store.base import VectorStoreConfig
+import langroid as lr
 
 
-class CustomAgentConfig(AgentConfig):
+class CustomAgentConfig(lr.AgentConfig):
     max_tokens: int = 10000
-    vecdb: VectorStoreConfig = None
-    llm: OpenAIGPTConfig = OpenAIGPTConfig(
-        type="openai",
-        chat_model=OpenAIChatModel.GPT4,
-        use_chat_for_completion=True,
-        cache_config=RedisCacheConfig(fake=False),
+    vecdb: Optional[lr.vector_store.base.VectorStoreConfig] = None
+    llm: lr.language_models.openai_gpt.OpenAIGPTConfig = (
+        lr.language_models.OpenAIGPTConfig(
+            type="openai",
+            chat_model=lr.language_models.OpenAIChatModel.GPT4,
+            use_chat_for_completion=True,
+            cache_config=lr.cachedb.redis_cachedb.RedisCacheConfig(fake=False),
+        )
     )
-    parsing: ParsingConfig = ParsingConfig()
+    parsing: lr.parsing.parser.ParsingConfig = lr.parsing.parser.ParsingConfig()
```

Comment:
So now the import langroid is a single line, which is great. Now the uses of `lr....` become longer, which I don't mind it's still a great improvement. You had earlier proposed setting things up so that the most common imports are available at "top level", e.g. `from langroid import AgentConfig` -- (a) is that even possible  (I think using "all" in `__init__` )? and (b) is this best done in a later PR? 
I think best to address in another PR

Reply:
The most common `Agent` classes and `Task` are already at top-level as they are likely to be used very frequently; however, other common classes (e.g. `OpenAIGPT`) aren't currently. We can easily add more. 

Alternatively, we could have the standard usage be `from langroid import agent` and avoid the top-level re-export; what do you think?

Right now, I'm not using __all__ but we can definitely make things `import *` friendly.

To make the `lr...` chains shorter, we can also export more symbols to the top of their respective submodules.

Most of this can wait for later; however, we do need to decide what needs to *not* be re-exported with this PR.

Reply:
> most common `Agent` classes and `Task` are already at top-level

Right I missed that.

> Alternatively, we could have the standard usage be `from langroid import agent` and avoid the top-level re-export; what do you think?

Confused here. My understanding is `from langroid import Agent` is accomplished via the top-level re-export you already have. Maybe I misunderstand the terminology.

> we do need to decide what needs to _not_ be re-exported with this PR.

This also I'm not understaning. We can choose to re-export some classes/fns, but leave out others. Why do we need to decide what "not" to re-export?

I agree we can address `__all__` later.

Reply:
> Alternatively, we could have the standard usage be `from langroid import agent` and avoid the top-level re-export; what do you think?

I think I see what you're saying -- instead of `import langroid as lr` and then use it as `lr.Agent` etc, you're saying we can suggest standard usage is `from langroid import Agent`. Actually I like the `lr.Agent` approach.

Reply:
The standard usage for some libraries (e.g. Pytorch) is to import both the top-level module and some submodules, i.e. `import torch` and `import torch.nn as nn`, If we choose not to expose them at the top level, leaving the `Agent` classes accessible only via `agent`, we could use that type of convention.

Currently, we can access these classes multiple ways, i.e. either `lr.Agent` or `lr.agent.Agent` or `lr.agent.base.Agent`; by the "needing to choose what not to re-export" I meant that removing any place a given symbol is exposed could break code, but adding more (e.g. by exposing more classes at the top level) will maintain compatibility.

Reply:
> "needing to choose what not to re-export" I meant that removing any place a given symbol is exposed could break code,

Got it. So conservatively exposing only a few key symbols like you did is fine. Later we can add more.